### PR TITLE
Add java-storage implementation for update items method.

### PR DESCRIPTION
**Pull request by shilpi23pandey**

### CODE REVIEW

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java [hunk 1]
```diff
@@ -494,6 +500,98 @@ public void close() {
...
+
+    if (itemInfoList.isEmpty()) {
+      return new ArrayList<>();
+    }
+
+    Map<StorageResourceId, GoogleCloudStorageItemInfo> resultItemInfos = new ConcurrentHashMap<>();
+    Set<IOException> innerExceptions = newConcurrentHashSet();
+    BatchExecutor executor = new BatchExecutor(storageOptions.getBatchThreads());
+
+    for (UpdatableItemInfo itemInfo : itemInfoList) {
```
Comment (arunkumarchacko) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java:516]: let us do this before creating executors etc.

**Review by arunkumarchacko — APPROVED** [review]

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java (hunk 1 above)
Comment (shilpi23pandey) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java:512]: Done

**Pull request merged**